docs/developer_guide.md

```diff
@@ -1,8 +1,147 @@
 # Developer Guide
 
-Developer Guide is coming soon, here are some resources:
+RAI is a problem-agnostic framework designed to provide a general solution for various tasks. It is easily extendable, allowing developers to adapt and integrate new functionalities and tools to meet specific needs.
+
+Before you dive into the developer guide, explore the idea behind RAI by helping the cow escape the farm with the following demo snippet!
+
+## Quickstart snippet
+
+```python
```

Comment:
Could we opt for a minimal example (much like a minimal_subscriber)?

I like the example but not as the first entry in the developer guide - maybe a link to "get a ready snippet to start early". This could also be in our examples - divided into files. Let's discuss.

Reply:
Perhaps the first chapter should be: try out / learn RAI by playing with the example

Reply:
I've simplified the example